File: Mercadona-api/Models/Product.cs
namespace Mercadona.Models;

public class Product
{
    public int ProductId { get; set; }
    public string Label { get; set; }
    public string Description { get; set; } 
    public float Price { get; set; }
    public string ImagePath { get; set; }


    public virtual Category Category { get; set; }  

    public virtual Promotion? Promotion { get; set; }   
}

File: Mercadona-ui/src/app/models/Product.ts
export class Product {
  productId!: number;
  label!: string;
  description!: string;
  price!: number;
  image!: string;
  category!: Category;
  promotion!: Promotion;
}

export class Category {
  categoryId!: number;
  label!: string;
}

export class Promotion {
  promotionId: number;
  start: Date;
  end: Date;
  discount: number;

  constructor(start: Date, end: Date, discount: number) {
    this.promotionId = 0;
    this.start = start;
    this.end = end;
    this.discount = discount;
  }
}
